ecl/hql/hqlutil.cpp

```diff
@@ -2645,6 +2645,49 @@ void checkSelectConsistency(IHqlExpression * expr)
 
 //---------------------------------------------------------------------------
 
+static HqlTransformerInfo parameterDependencyCheckerInfo("ParameterDependencyChecker");
+class ParameterDependencyChecker  : public NewHqlTransformer
+{
+public:
+    ParameterDependencyChecker() : NewHqlTransformer(parameterDependencyCheckerInfo), foundParameter(false)
+    {
+    }
+
+    virtual void analyseExpr(IHqlExpression * expr)
+    {
+        if (expr->isFullyBound() || alreadyVisited(expr) || foundParameter)
```

Comment:
alreadyVisited? I presume this is here to account for the logic that if it has been visited then this function (analyseExpr via isDependentOnParameter) has already been called, however, is it assured, i.e. that every 'visitation' i.e. anything that increases the visitation (pass) count, also calls isDependentOnParameter?